lib/plug/conn.ex

```diff
@@ -590,13 +623,20 @@ defmodule Plug.Conn do
 
   It is recommended for header keys to be in lower-case, to avoid sending
   duplicate keys in a request. As a convenience, this is validated during
-  testing which raises a `Plug.Conn.InvalidHeaderError` if the header key
-  is not lowercase.
+  testing where a header that is not lowercase raises a
```

Comment:
I found this statement (and the previous version) a bit confusing. Could we try:

 > As a convenience, when using the `Plug.Adapters.Conn.Test` adapter, any headers that aren't lowercase will raise a `Plug.Conn.InvalidHeaderError`